src/App.tsx

```diff
@@ -17,18 +53,36 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                setQuery={setQuery}
+                filter={filter}
+                setFilter={setFilter}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length > 0
+                && (
+                  <TodoList
+                    todos={preparedTodos}
+                    selectedTodo={selectedTodo}
+                    onSelectedTodo={setSelectedTodo}
+                  />
+                )}
+
+              {isTodoLoading && <Loader />}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      { selectedTodo && (
+        <TodoModal
+          selectedTodo={selectedTodo}
+          deleteSelectedTodo={setSelectedTodo}
```

Comment:
Weird name. You don't delete this todo, you just unselect it.
You can name it setSelectedTodo or unselectTodo in case when this function can make only unselect operation, but it doesn't